flink-ml-iteration/src/main/java/org/apache/flink/iteration/proxy/state/ProxyOperatorStateBackend.java

```diff
@@ -58,12 +62,24 @@ public <K, V> BroadcastState<K, V> getBroadcastState(MapStateDescriptor<K, V> st
         return wrappedBackend.getBroadcastState(newDescriptor);
     }
 
+    private <S> TypeInformation<S> getTypeInfo(ListStateDescriptor<S> stateDescriptor)
+            throws Exception {
+        Field f = StateDescriptor.class.getDeclaredField("typeInfo");
+        f.setAccessible(true);
+        return ((ListTypeInfo) f.get(stateDescriptor)).getElementTypeInfo();
+    }
+
     @Override
     public <S> ListState<S> getListState(ListStateDescriptor<S> stateDescriptor) throws Exception {
         ListStateDescriptor<S> newDescriptor =
-                new ListStateDescriptor<>(
-                        stateNamePrefix.prefix(stateDescriptor.getName()),
-                        stateDescriptor.getElementSerializer());
+                stateDescriptor.isSerializerInitialized()
```

Comment:
Does `getUnionListState()` also need to check this?

Reply:
Good catch. I have updated the check for `UnionListState` and `BroadcastState`.